quint/src/runtime/impl/compilerImpl.ts

```diff
@@ -1325,7 +1348,7 @@ export class CompilerVisitor implements IRVisitor {
               },
             }
           })
-          return this.chainAllOrThen(actions, 'then')
+          return this.chainAllOrThen(actions, 'then', _ => app.args[1].id)
```

Comment:
```suggestion
          // In case the case of reps, we have multiple copies of the same action. This is why all occurrences have the same id.
          return this.chainAllOrThen(actions, 'then', _ => app.args[1].id)
```